### PR TITLE
* Changes direct usages of GUI.enabled to instead use EditorGUI.Begin…

### PR DESCRIPTION
**Pull request by MunchyYDL**
…DisabledGroup() / EditorGUI.EndDisabledGroup()

### Requirements

* Filling out the template is required. Any pull request that does not include enough information to be reviewed in a timely manner may be closed at the maintainers' discretion.
* All new code requires tests to ensure against regressions

### Description of the Change

I have replaced all the possible direct usages of **GUI.enabled** with a corresponding set of calls to the **EditorGUI.BeginDisabledGroup()** and **EditorGUI.EndDisabledGroup()**. I have also used the code block style that I see you have used before, which places the code between the calls in a separate code block with { }, and indents the code making it easier to read.

I have also done two small other changes in this PR.

1. Renamed the file PUblishView.cs to PublishView.cs
2. Renamed the variable busy to isBusy in AuthenticationView.cs, to follow the naming in the other views.

### CODE REVIEW

**Comment by StanleyGoldman** [comment]
Hey @MunchyYDL thanks for the pull request!

**Comment by StanleyGoldman** [comment]
Eek, some of your changes in `PUblishView.cs` might conflict with some changes I have in #196.
But otherwise it's looking good initially.

File: src/UnityExtension/Assets/Editor/GitHub.Unity/UI/PublishView.cs [hunk 1]
```diff
@@ -132,10 +132,11 @@ public override void OnGUI()
                 GUILayout.BeginVertical();
                 {
                     GUILayout.Label("Owner");
-
-                    GUI.enabled = !isBusy;
-                    selectedOwner = EditorGUILayout.Popup(0, owners);
-                    GUI.enabled = true;
+                    EditorGUI.BeginDisabledGroup(isBusy);
```
Comment (StanleyGoldman) [inline src/UnityExtension/Assets/Editor/GitHub.Unity/UI/PublishView.cs]: We can use a wider scope here.

File: src/UnityExtension/Assets/Editor/GitHub.Unity/UI/HistoryView.cs [hunk 2]
```diff
@@ -207,7 +207,8 @@ private void RefreshLog()
             {
                 if (Repository != null)
                 {
-                    GitClient.Log().ThenInUI((success, log) => {
+                    GitClient.Log().ThenInUI((success, log) =>
+                    {
```
Comment (StanleyGoldman) [inline src/UnityExtension/Assets/Editor/GitHub.Unity/UI/HistoryView.cs]: I think you accidentally re-formatted this chunk of code here, could you restore it?

File: src/UnityExtension/Assets/Editor/GitHub.Unity/UI/HistoryView.cs [hunk 3]
```diff
@@ -569,7 +585,8 @@ private void RevertCommit()
             {
                 Repository
                     .Revert(selection.CommitID)
-                    .FinallyInUI((success, e) => {
+                    .FinallyInUI((success, e) =>
+                    {
```
Comment (StanleyGoldman) [inline src/UnityExtension/Assets/Editor/GitHub.Unity/UI/HistoryView.cs]: Same here.

File: src/UnityExtension/Assets/Editor/GitHub.Unity/UI/HistoryView.cs [hunk 4]
```diff
@@ -699,7 +716,8 @@ private void Pull()
                             // (either git rebase --abort or git merge --abort)
                         }
                     }, true)
-                    .FinallyInUI((success, e) => {
+                    .FinallyInUI((success, e) =>
+                    {
```
Comment (StanleyGoldman) [inline src/UnityExtension/Assets/Editor/GitHub.Unity/UI/HistoryView.cs]: Same here.

File: src/UnityExtension/Assets/Editor/GitHub.Unity/UI/HistoryView.cs [hunk 5]
```diff
@@ -722,7 +740,8 @@ private void Push()
             var remote = Repository.CurrentRemote.HasValue ? Repository.CurrentRemote.Value.Name : String.Empty;
             Repository
                 .Push()
-                .FinallyInUI((success, e) => {
+                .FinallyInUI((success, e) =>
+                {
```
Comment (StanleyGoldman) [inline src/UnityExtension/Assets/Editor/GitHub.Unity/UI/HistoryView.cs]: Same here.

File: src/UnityExtension/Assets/Editor/GitHub.Unity/UI/HistoryView.cs [hunk 6]
```diff
@@ -744,7 +763,8 @@ private void Fetch()
             var remote = Repository.CurrentRemote.HasValue ? Repository.CurrentRemote.Value.Name : String.Empty;
             Repository
                 .Fetch()
-                .FinallyInUI((success, e) => {
+                .FinallyInUI((success, e) =>
+                {
```
Comment (StanleyGoldman) [inline src/UnityExtension/Assets/Editor/GitHub.Unity/UI/HistoryView.cs]: Same here :trollface: 

... Please don't hate me 🙈

File: src/UnityExtension/Assets/Editor/GitHub.Unity/UI/HistoryView.cs (hunk 2 above)
Comment (MunchyYDL) [inline src/UnityExtension/Assets/Editor/GitHub.Unity/UI/HistoryView.cs]: Sure, will fix that. I use tabs for spacing normally, so I had to reformat the code after I had done my changes, and thought that you guys just had missed this piece of formatting before. I will revert it. :)

File: src/UnityExtension/Assets/Editor/GitHub.Unity/UI/HistoryView.cs (hunk 6 above)
Comment (MunchyYDL) [inline src/UnityExtension/Assets/Editor/GitHub.Unity/UI/HistoryView.cs]: Haha, no worries, I understand that you want this fixed.

File: src/UnityExtension/Assets/Editor/GitHub.Unity/UI/AuthenticationView.cs [hunk 7]
```diff
@@ -134,35 +134,42 @@ private void OnGUILogin()
             GUILayout.Space(3);
             GUILayout.BeginHorizontal();
             {
-                if (busy) GUI.enabled = false;
-                username = EditorGUILayout.TextField(usernameLabel ,username, Styles.TextFieldStyle);
-                GUI.enabled = true;
+                EditorGUI.BeginDisabledGroup(isBusy);
```
Comment (MunchyYDL) [inline src/UnityExtension/Assets/Editor/GitHub.Unity/UI/AuthenticationView.cs]: I will look at this as well, didn't want to change scope before, as I didn't know what you guys wanted.

**Comment by MunchyYDL** [comment]
So, I have made some changes, can you please have a look and see if this is more in line with what you wanted? :)

**Comment by StanleyGoldman** [comment]
Hey @MunchyYDL sorry to leave you hanging. I will take a look at your stuff today.

_I also sent you an email to the one you've listed on your profile_

**Comment by MunchyYDL** [comment]
No worries @StanleyGoldman, I had some other things to work on as well. :)

File: src/UnityExtension/Assets/Editor/GitHub.Unity/UI/AuthenticationView.cs [hunk 8]
```diff
@@ -131,81 +131,83 @@ private void HandleEnterPressed()
...
-            if (GUILayout.Button(loginButton) || (GUI.enabled && enterPressed))
-            {
-                GUI.FocusControl(null);
-                busy = true;
-                AuthenticationService.Login(username, password, DoRequire2fa, DoResult);
+                GUILayout.Space(Styles.BaseSpacing + 3);
+                GUILayout.BeginHorizontal();
+                {
+                    GUILayout.FlexibleSpace();
+                    if (GUILayout.Button(loginButton) || (GUI.enabled && enterPressed))
```
Comment (StanleyGoldman) [inline src/UnityExtension/Assets/Editor/GitHub.Unity/UI/AuthenticationView.cs]: Hmm, I think we have to do something about this `GUI.enabled` here.

File: src/UnityExtension/Assets/Editor/GitHub.Unity/UI/AuthenticationView.cs [hunk 9]
```diff
@@ -131,81 +131,83 @@ private void HandleEnterPressed()
...
+                    {
+                        GUILayout.FlexibleSpace();
+                        if (GUILayout.Button(backButton))
+                        {
+                            GUI.FocusControl(null);
+                            need2fa = false;
+                            Redraw();
+                        }
+
+                        if (GUILayout.Button(twofaButton) || (GUI.enabled && enterPressed))
```
Comment (StanleyGoldman) [inline src/UnityExtension/Assets/Editor/GitHub.Unity/UI/AuthenticationView.cs]: Same scenario here.

File: src/UnityExtension/Assets/Editor/GitHub.Unity/UI/HistoryView.cs [hunk 10]
```diff
@@ -311,16 +311,19 @@ private void DoOfferToInitializeRepositoryGUI()
...
+                    if (GUILayout.Button(Localization.InitializeRepositoryButtonText, "Button"))
+                    {
+                        isBusy = true;
+                        Manager.InitializeRepository()
+                            .FinallyInUI(() => isBusy = false)
+                            .Start();
+                    }
                 }
+                EditorGUI.EndDisabledGroup();
                 GUI.enabled = enabled;
```
Comment (StanleyGoldman) [inline src/UnityExtension/Assets/Editor/GitHub.Unity/UI/HistoryView.cs]: This should go away.

File: src/UnityExtension/Assets/Editor/GitHub.Unity/UI/HistoryView.cs [hunk 11]
```diff
@@ -311,16 +311,19 @@ private void DoOfferToInitializeRepositoryGUI()
                 GUILayout.FlexibleSpace();
 
                 var enabled = GUI.enabled;
```
Comment (StanleyGoldman) [inline src/UnityExtension/Assets/Editor/GitHub.Unity/UI/HistoryView.cs]: This variable should go away too,

**Comment by MunchyYDL** [comment]
So, now I had some time to look at this again!

Wasn't aware of you pushing to my repo, I hope that doesn't mess up anything about the history in my branch, as I didn't notice it before I had done my last commit locally. :)

## AuthenticationView.cs
Didn't think this through I guess, but now I have replaced **GUI.enabled** with a check for **!isBusy**, to prevent multiple invocations of the button handlers in both the "Normal" and "2FA" login code.

## HistoryView.cs
What can I say? Well, it's gone now. :)

**Comment by MunchyYDL** [comment]
Oh, and one more thing, what's the difference of the two appveyor checks?

As you can see, I'm a bit new to this whole PR thing, so I hope I'm not giving you too much trouble in "helping" you guys with this. :)

**Comment by shana** [comment]
@MunchyYDL One of the builds is for your branch, and the other is for your branch merged with the target branch (master in this case). Since your branch is in a forked repo, AppVeyor isn't running a build for it, and it only does the combined branch+master build. I just can't convince the UI to ignore this check if the branch is on a fork so it makes for a bit of confusing status checks, but it isn't a problem.

**Comment by shana** [comment]
> Wasn't aware of you pushing to my repo, I hope that doesn't mess up anything about the history in my branch, as I didn't notice it before I had done my last commit locally. :)

Ah, yeah, sorry, that's just a thing we're used to doing, updating branches with the latest master so it's all up to date, it's kind of routine for all branches (they can't be merged if they're not up to date), but we shouldn't do it for branches we don't own without letting you know first (and preferably you do it, not us). I hope that didn't cause any issues on your end!

**Comment by StanleyGoldman** [comment]
We are happy to help... but also, you started helping first! 🍻 

There are a lot of reasons why.. Mostly, we can't fix everything.
And this time you are doing something simple, next time it will be a bit harder.

Hopefully soon you'll be taking whole features off our hands.. 😈

**Comment by StanleyGoldman** [comment]
FYI, there is a checkbox when you were creating the pull request that allows us to push just to that branch in your fork.

https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/

**Comment by MunchyYDL** [comment]
Thanks for your help and nice comments!

I've been using both Git and Unity before, so I thought this project was a nice combination to help me further my understanding of both, and be able to submit my first PR to an open project.

**Pull request merged**